include/natalie/random_value.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <utility>
+
+#include "natalie/class_value.hpp"
+#include "natalie/forward.hpp"
+#include "natalie/global_env.hpp"
+#include "natalie/integer_value.hpp"
+#include "natalie/value.hpp"
+#include <random>
+
+namespace Natalie {
+
+using namespace TM;
+
+class RandomValue : public Value {
+public:
+    RandomValue(ClassValue *klass)
+        : Value { Value::Type::Random, klass } { }
+
+    RandomValue()
+        : Value { Value::Type::Random, GlobalEnv::the()->Random() } { }
+
+    ~RandomValue() { free(m_generator); }
+
+    virtual void visit_children(Visitor &visitor) override final {
+        Value::visit_children(visitor);
+    }
+
+    ValuePtr initialize(Env *, ValuePtr);
+
+    ValuePtr rand(Env *, ValuePtr);
+    ValuePtr seed() { return new IntegerValue { m_seed }; }
```

Comment:
```suggestion
    ValuePtr seed() { return ValuePtr::integer(m_seed); }
```

`ValuePtr::integer` doesn't cause an allocation and can save some memory in certain optimized circumstances.